exasol_script_languages_developer_sandbox/lib/run_setup_ec2_and_install_dependencies.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import signal
+import traceback
+from typing import Tuple, Optional
+
+from exasol_script_languages_developer_sandbox.lib.ansible.ansible_access import AnsibleAccess, AnsibleException
+from exasol_script_languages_developer_sandbox.lib.ansible.ansible_repository import AnsibleRepository, \
+    default_repositories
+from exasol_script_languages_developer_sandbox.lib.ansible.ansible_run_context import AnsibleRunContext, \
+    default_ansible_run_context
+from exasol_script_languages_developer_sandbox.lib.aws_access import AwsAccess
+
+from exasol_script_languages_developer_sandbox.lib.host_info import HostInfo
+
+from exasol_script_languages_developer_sandbox.lib.run_install_dependencies import run_install_dependencies
+from exasol_script_languages_developer_sandbox.lib.run_setup_ec2 import run_lifecycle_for_ec2
+
+
+def run_setup_ec2_and_install_dependencies(aws_access: AwsAccess,
+                                           ec2_key_file: Optional[str], ec2_key_name: Optional[str],
+                                           ansible_access: AnsibleAccess,
+                                           ansible_run_context: AnsibleRunContext = default_ansible_run_context,
+                                           ansible_repositories: Tuple[AnsibleRepository, ...] = default_repositories
+                                           ) -> None:
+    """
+    Runs setup of an EC2 instance and then installs all dependencies via Ansible.
+    Note that if an error occurs during the ansible installation, the EC2-machine is not removed immediately. This
+    gives you time to login into the machine and identify any setup issues.
+    You can stop the EC-2 machine by pressing Ctrl-C.
```

Comment:
Nice, however this is not usable in automated usage

Reply:
Yeah, the purpose of this is just debugging....